pkg/pillar/cmd/watcher/watcher.go

```diff
@@ -268,11 +529,17 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	// Handle memory pressure events by calling GC explicitly
 	go handleMemoryPressureEvents()
 
+	// Set the goroutine leak detection parameters to default values
+	ctx.GRLDParams.Set(5000, 1*time.Minute, 10*time.Minute, 24*time.Hour, 5*time.Minute)
```

Comment:
You don't need this; a readGlobalGoroutineLeakDetectionParams() is sufficient.

You have already waited for GCinitialized, so all of the ConfigItems are known.
And the upgradeconverter (or is it zedagent) ensures that all new ConfigItems are filled in with their default values before the ConfigItems are published.

Reply:
Removed!